plugins/xmlimport/value.c

```diff
@@ -0,0 +1,319 @@
+            break;
+        case VALUE_STATE_NODEID:
+            break;
+    }
+}
+
+static void
+setBoolean(uintptr_t adr, char *value) {
+    *(UA_Boolean *)adr = isTrue(value);
+}
+
+static void
+setSByte(uintptr_t adr, char *value) {
+    *(UA_SByte *)adr = (UA_SByte)atoi(value);
+}
+
+static void
+setByte(uintptr_t adr, char *value) {
+    *(UA_Byte *)adr = (UA_Byte)atoi(value);
+}
+
+static void
+setInt16(uintptr_t adr, char *value) {
+    *(UA_Int16 *)adr = (UA_Int16)atoi(value);
+}
+
+static void
+setUInt16(uintptr_t adr, char *value) {
+    *(UA_UInt16 *)adr = (UA_UInt16)atoi(value);
+}
+
+static void
+setInt32(uintptr_t adr, char *value) {
+    *(UA_Int32 *)adr = atoi(value);
+}
+
+static void
+setUInt32(uintptr_t adr, char *value) {
+    *(UA_UInt32 *)adr = (UA_UInt32)atoi(value);
+}
+
+static void
+setInt64(uintptr_t adr, char *value) {
+    UA_assert(false && "not implemented");
+}
+
+static void
+setUInt64(uintptr_t adr, char *value) {
+    UA_assert(false && "not implemented");
+}
+
+static void
+setFloat(uintptr_t adr, char *value) {
+    *(UA_Float *)adr = (UA_Float)atof(value);
+}
+
+static void
+setDouble(uintptr_t adr, char *value) {
+    *(UA_Double *)adr = atof(value);
+}
+
+static void
+setString(uintptr_t adr, char *value) {
+    UA_String *s = (UA_String *)adr;
+    s->length = strlen(value);
+    s->data = (UA_Byte *)value;
+}
+
+static void
+setLocalizedText(uintptr_t adr, char *value) {
+    UA_LocalizedText *s = (UA_LocalizedText *)adr;
+    s->locale = UA_STRING_NULL;
+    s->text.data = (UA_Byte *)value;
+    s->text.length = strlen(value);
+}
+
+static void
+setDateTime(uintptr_t adr, char *value) {
+    printf("DateTime: %s currently not handled\n", value);
+}
+
+static void
+setNodeId(uintptr_t adr, char *value) {
+    // todo: namespaceIndex should be converted?
+    *(UA_NodeId *)adr = extractNodeId(value);
+}
+
+static void
+setNotImplemented(uintptr_t adr, char *value) {
+    UA_assert(false && "not implemented");
+}
+
+static const ConversionFn conversionTable[UA_DATATYPEKINDS] = {
+    setBoolean,        setSByte,          setByte,           setInt16,
+    setUInt16,         setInt32,          setUInt32,         setInt64,
+    setUInt64,         setFloat,          setDouble,         setString,
+    setDateTime,       setNotImplemented,
+    setString,  // handle like bytestring like string
+    setNotImplemented, setNodeId,         setNotImplemented, setNotImplemented,
+    setNotImplemented, setLocalizedText,  setNotImplemented, setNotImplemented,
+    setNotImplemented, setNotImplemented, setNotImplemented,
+    setInt32,  // handle enum the same way like int32
+    setNotImplemented, setNotImplemented, setNotImplemented, setNotImplemented};
+
+static void
+setScalarValue(Value *val, const UA_DataType *type, size_t padding, char *value) {
+    uintptr_t adr = (uintptr_t)val->value + val->offset + padding;
+    if(value) {
+        conversionTable[type->typeKind](adr, value);
```

Comment:
Overflow check for type->typeKind missing?

Reply:
think not necessary here, typekind should be set to one of the enum values in types.h.